### ISSUE TITLE
bug: messages don't send from FFI

### ISSUE THREAD

**Issue by juliettebee** [matrix-org/matrix-rust-sdk#1242]
I'm trying to send messages from FFI here's my code: 
```swift
let listner = Listener(onUpdate: { diff in
      print(diff.push()!.fmtDebug())
})
room.wrappedValue.addTimelineListener(listener: listner)
try room.wrappedValue.send(msg: messageEventContentFromMarkdown(md: "Test"), txnId: genTransactionId())
room.wrappedValue.removeTimeline()
```

When this gets called it prints the TimelineDiff but nothing shows in Element or any other client. The method `send(msg: txnID:)` never returns and no error is thrown.

**Comment by jplatte** [comment]
Yeah, we've been seeing this in ElementX too. It doesn't happen all the time. Currently trying to get more info to narrow down the issue.

**Comment by jplatte** [comment]
This might be due to the SDK trying to figure out whether the room is encrypted first by asking the server for the room's `m.room.encrypted` event, and accidentally retrying a 404 response forever. If that is the issue, it should be fixed as of #1237. Can you check?

**Comment by juliettebee** [comment]
This fixes my issue, thank you!
